clients/go/admin/token_source_provider.go

```diff
@@ -45,16 +50,21 @@ func NewTokenSourceProvider(ctx context.Context, cfg *Config, tokenCache cache.T
 			tokenURL = metadata.TokenEndpoint
 		}
 
+		clientMetadata, err := authClient.GetPublicClientConfig(ctx, &service.PublicClientAuthConfigRequest{})
```

Comment:
should we only fetch this if the `cfg.Audience` is empty?

Reply:
Done., if either scopes or Audience is empty then we call this api